### PR TITLE
[TIR] Introduce BufferLoad/Store

### PR DESCRIPTION
**Pull request by tqchen**
Co-authored-by: Siyuan Feng <hzfengsy@sjtu.edu.cn>

This PR introduces BufferLoad/Store to TIR. The new nodes will replace
Provide and Call with Tensor arguments in the subsequent refactors.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @Hzfengsy @yzhliu @ZihengJiang @spectrometerHBH

**Comment by Hzfengsy** [comment]
Did you push the wrong code to this branch? The codes are same as another PR

**Comment by tqchen** [comment]
@Hzfengsy sorry, just fixed

File: include/tvm/tir/var.h [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+   */
+  kOrdered = 3,
+  /*!
+   * \brief IterVar is opaque,
+   *
+   *  May not corresponds to any generated loop
+   *  Disallow all IterVar manipulations and compute_at
+   *
+   * \note This is usually used to implement composite op
+   *  or external op, where the
```
Comment (TH3CHARLie) [inline include/tvm/tir/var.h:217]: the comment here seems incomplete, not sure about what kind of constraints these ops have led to opaque iter var.

**Pull request merged**